sqlx-core/src/postgres/protocol/authentication.rs

```diff
@@ -86,12 +95,34 @@ impl Decode for Authentication {
             }
 
             11 => {
-                let mut data = Vec::with_capacity(buf.len());
-                data.extend_from_slice(buf);
-
-                Authentication::SaslContinue {
-                    data: data.into_boxed_slice(),
-                }
+                let mut salt: Vec<u8> = Vec::new();
+                let mut nonce: Vec<u8> = Vec::new();
+                let mut iter_count: u32 = 0;
+
+                buf.split(|byte| *byte == b',')
+                    .map(|s| {
+                        let (key, value) = s.split_at(1);
+                        let value = value.split_at(1).1;
+
+                        (key[0] as char, value)
+                    })
+                    .for_each(|(key, value)| match key {
+                        's' => salt = value.to_vec(),
+                        'r' => nonce = value.to_vec(),
+                        'i' => {
+                            iter_count = u32::from_str_radix(&String::from_utf8_lossy(&value), 10)
+                                .unwrap_or(0);
+                        }
+
+                        _ => {}
+                    });
+
+                Authentication::SaslContinue(SaslContinue {
+                    salt: base64::decode(&salt).unwrap(),
+                    nonce,
+                    iter_count,
+                    data: String::from_utf8_lossy(buf).into_owned(),
```

Comment:
Don't use `_lossy` and instead just error. We should have examples of similar elsewhere.